cli/asc.js

```diff
@@ -47,7 +47,7 @@ const find = require("./util/find");
 const binaryen = global.binaryen || (global.binaryen = require("binaryen"));
 
 const dynrequire = typeof __webpack_require__ === "function"
-  ? __non_webpack_require__
+  ? eval('require')
```

Comment:
One reason for making the switch that comes to my mind is that using `eval` may not be allowed in certain environments, for example if `eval` is disabled by CSP. Iirc there are setups where such a JS file would not work at all, while other environments may print a warning. At the very least we need a try-catch around `eval` I think.

Reply:
You are right, this won't fit like this. And when I think about it, the main reason may just be that it will not work at all, because `require` being undefined (as in "not defined at all"), this `eval` will throw instead of returning undefined.

Forget this for now, I'll rework all this by using `__non_webpack_require__` WITHOUT letting webpack detect that `dynrequire ===  __non_webpack_require__`, that would solve the problem.